frontend/src/app/libs/common-components/components/input/input.component.ts

```diff
@@ -0,0 +1,229 @@
+
+	private syncValidators: ValidatorFn[] | null = null;
+	private asyncValidators: AsyncValidatorFn[] | null = null;
+	private defaultValue: number | string | boolean | null | undefined;
+	private ngUnsubscribe: Subject<any> = new Subject();
+
+	public get isDarkTheme(): boolean {
+		return this.theme === ComponentTheme.Dark;
+	}
+
+	public get isText(): boolean {
+		return this.type === InputType.Text ? true : false;
+	}
+
+	public get isEmail(): boolean {
+		return this.type === InputType.Email ? true : false;
+	}
+
+	public get isPassword(): boolean {
+		return this.type === InputType.Password ? true : false;
+	}
+
+	public get inputType(): string {
+		if (this.type !== 'password') {
+			return this.type;
+		} else if (this.type === 'password' && this.hide) {
+			return 'password';
+		} else if (this.type === 'password' && !this.hide) {
+			return 'text';
+		}
+	}
+	public constructor(
+		@Optional() @Host() @SkipSelf() protected parentFormContainer: ControlContainer,
+		public elementRef: ElementRef,
+		public changeDetector: ChangeDetectorRef
+	) {}
+
+	public ngOnInit(): void {
+		this.defaultValue = Boolean(this.value) ?  this.value : null;
+		this.formControlName = this.elementRef.nativeElement.getAttribute('formControlName');
+		if (this.formControlName != null && this.parentFormContainer != null) {
+
+			this.formControl = (this.parentFormContainer.control as FormGroup).controls[this.formControlName] as FormControl;
+			if (this.formControl === undefined) {
+				throw new Error(`Form control ${this.formControlName} is not registered in form group`);
+			}
+
+		} else {
+			this.formControl = new FormControl('');
+		}
+
+		this.subscribeToChanges();
+
+		this.inlineSyncValidators = this.getInlineSyncValidatorsFromFormControl();
+		this.inlineAsyncValidators = this.getInlineAsyncValidatorsFromFormControl();
+		this.setSyncValidator();
+		this.setAsyncValidator();
+		this.setValidatorsToFormControl();
+		this.switchToDisable();
+	}
+
+	public ngAfterViewInit(): void {
+		if (Boolean(this.defaultValue)) {
+			setTimeout(() => {
+					this.formControl.setValue(this.defaultValue);
+					this.onTouched();
+			});
+		}
+	}
+
+	public ngOnDestroy(): void {
+		this.ngUnsubscribe.next();
+		this.ngUnsubscribe.complete();
+	}
+
+	public switchToDisable(): void {
+		if (Boolean(this.formControl) && this.disabled) {
+			this.formControl.disable();
+		}
+	}
+	public switchToEnable(): void {
+		if (Boolean(this.formControl) && this.disabled) {
+			this.formControl.enable();
+		}
+	}
+
+	public onTouched: any = () => undefined;
+
+	public onChange: any = () => undefined;
+
+	public writeValue(value: any): void {
+		this.value = value;
+		this.onChange(value);
+		if (this.formControl && this.formControl.value) {
+			this.onTouched();
+		}
+		this.valueChange.emit(value);
+	}
+
+	public onValueChange(value: any): void {
+		this.onTouched();
+		if (Boolean(this.value) && this.value !== value) {
+			this.value = value;
+			this.onChange(value);
```

Comment:
this line duplicates frequently, please move to separate method and replace all occurrences by using method